Core/src/IfcGeometryConverter/SplineUtilities.h

```diff
@@ -0,0 +1,768 @@
+/*
+	Copyright (c) 2018 Technical University of Munich
+	Chair of Computational Modeling and Simulation.
+
+	TUM Open Infra Platform is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License Version 3
+	as published by the Free Software Foundation.
+
+	TUM Open Infra Platform is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// visual studio
+#pragma once
+// linux
+#ifndef SPLINEUTILITIES_H
+#define SPLINEUTILITIES_H
+
+#include <vector>
+#include <tuple>
+#include <utility>
+#include <cmath>
+#include "CarveHeaders.h"
+
+namespace OpenInfraPlatform
+{
+	namespace Core {
+		namespace IfcGeometryConverter {
+			/*! \brief Provides some functions for spline calculations.
+			 *
+			 * Here are functions which are used from the classes \c SplineConverterT and \c SplineInterpretation [and maybe more in future].
+			 * These functions are independent from \c GeometrySettings, \c UnitConverter, etc., which aren't included in \c SplineInterpretation.
+			 */
+			class SplineUtilities
+			{
+			public:
+				SplineUtilities() {}
+
+				virtual ~SplineUtilities() {}
+
+				/*! \brief Computes the curve points of the B-Spline.
+				 *
+				 * All information has to be loaded from an \c IfcBSplineCurveWithKnots entity before calling this function.
+				 *
+				 * \param[in]	order				Order of the B-Spline or rather the basis functions ( =degree+1 )
+				 * \param[in]	knotArray			The array / vector of knots, the function \c loadKnotArray gives this vector.
+				 * \param[in]	controlPoints		The vector of the B-Spline control points.
+				 * \param[in]	numControlPoints	The total number of B-Spline control points ( =n+1 )
+				 * \param[in]	numCurvePoints		The number of curve points where the curve c(t) has to be evaluated.
+				 * \param[in]	accuracy			Accuracy which is technically needed in the calculation.
+				 *
+				 * \return		The array of curve points, which can be rendered in a viewport after correction by \c GeomUtils::appendPointsToCurve.
+				 */
+				 // B-Spline curve definition according to: http://mathworld.wolfram.com/B-Spline.html
+				std::vector<carve::geom::vector<3>> computeBSplineCurveWithKnots(
+					const int& order,
+					const std::vector<double>& knotArray,
+					const std::vector<carve::geom::vector<3>>& controlPoints,
+					const uint32_t& numCurvePoints,
+					const double& accuracy) const throw(...)

```

Comment:
For example, this can definitely be a `static` function.

Reply:
Done in commit 4f9851b.